File: spec/functional_test/fixtures/java/spring/src/WebSocketConfig.java
package com.example;

import org.springframework.context.annotation.Configuration;
import org.springframework.messaging.simp.config.MessageBrokerRegistry;
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker;
import org.springframework.web.socket.config.annotation.StompEndpointRegistry;
import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer;

@Configuration
@EnableWebSocketMessageBroker
public class WebSocketConfig implements WebSocketMessageBrokerConfigurer {
    private static final String APP_PREFIX = "/app";
    private static final String SOCKET_ROOT = "/socket";
    private static final String CHAT_PATH = "/chat/{roomId}";

    @Override
    public void registerStompEndpoints(StompEndpointRegistry registry) {
        registry.addEndpoint(SOCKET_ROOT + CHAT_PATH, "/portfolio").withSockJS();
    }

    @Override
    public void configureMessageBroker(MessageBrokerRegistry registry) {
        registry.setApplicationDestinationPrefixes(APP_PREFIX);
    }
}

File: spec/functional_test/fixtures/kotlin/spring_websocket/src/main/kotlin/com/example/WebSocketConfig.kt
package com.example

import org.springframework.context.annotation.Configuration
import org.springframework.messaging.simp.config.MessageBrokerRegistry
import org.springframework.web.socket.config.annotation.EnableWebSocketMessageBroker
import org.springframework.web.socket.config.annotation.StompEndpointRegistry
import org.springframework.web.socket.config.annotation.WebSocketMessageBrokerConfigurer

@Configuration
@EnableWebSocketMessageBroker
class WebSocketConfig : WebSocketMessageBrokerConfigurer {
    override fun registerStompEndpoints(registry: StompEndpointRegistry) {
        registry.addEndpoint(SOCKET_ROOT + CHAT_PATH, "/portfolio").setAllowedOrigins("*").withSockJS()
    }

    override fun configureMessageBroker(registry: MessageBrokerRegistry) {
        registry.setApplicationDestinationPrefixes(APP_PREFIX)
    }

    companion object {
        const val SOCKET_ROOT = "/socket"
        const val CHAT_PATH = "/chat/{roomId}"
        const val APP_PREFIX = "/app"
    }
}
